doc/callback.md

```diff
@@ -39,6 +41,23 @@ class Callback {
 
   void Reset();
 
+  v8::Local<v8::Value> Call(v8::Local<v8::Object> target,
```

Comment:
People can, and very commonly do, use a single `Nan::Callback` to repeatedly call the same function multiple times, potentially in difference asynchronous contexts. I think it would be dangerous, i.e. prone to misuse, if we associated an async resource with a `Callback` instance – given how this class is used / intended to be used.

From a convenience point of view, `AsyncWorker` provides a significantly more convenient API that handles async resource management automatically. I don't think we should make this class more convenient at the expense of making it dangerous.

Reply:
Some previous discussion on this topic: https://github.com/nodejs/node/issues/13254#issuecomment-357392244